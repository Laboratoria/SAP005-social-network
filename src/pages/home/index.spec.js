import { Home } from './index.js';
import * as services from '../../services/index.js';

describe('Home', () => {
  beforeEach(() => {
    services.loginWithEmail = jest.fn(() => Promise.resolve(true));
    services.loginGoogle = jest.fn(() => Promise.resolve(true));
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should be a function', () => {
    expect(typeof Home).toBe('function');
  });
  it('shoul return homePage', () => {
    expect(Home()).toMatchSnapshot();
  });
  it('should click submmit button', () => {
    const homeTest = Home();
    const btn = homeTest.querySelector('#submit-home');
    homeTest.querySelector('#email-input').value = 'email@teste1.com';
    homeTest.querySelector('#password-home').value = 'teste1';
    btn.dispatchEvent(new Event('click'));
    expect(services.loginWithEmail).toHaveBeenCalledWith('email@teste1.com', 'teste1');
  });
  it('should login with google', () => {
    const homeTest = Home();
    const autGoogle = homeTest.querySelector('#authGoogle');
    autGoogle.dispatchEvent(new Event('click'));
    expect(services.loginGoogle).toHaveBeenCalled();
  });
});
