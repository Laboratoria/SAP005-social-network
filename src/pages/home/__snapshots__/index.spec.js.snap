// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home shoul return homePage 1`] = `
<div
  class="form-container"
>
  
  
  <div
    class="form-login"
  >
    
  
    <div
      class="header-container"
    >
      
    
      <figure
        class="logo"
      >
        <img
          alt="Logo Runners"
          id="logo"
          src="./assets/logo/runners-360px.png"
        />
      </figure>
      
    
      <h1>
        Ol√°!
      </h1>
      
    
      <p
        class="about"
      >
        Conecte-se e compartilhe seus desafios com outros corredores.
      </p>
      
    
      <hr />
      
  
    </div>
    
  
    <div>
      
    
      <h2
        class="about"
      >
        Login
      </h2>
      
  
    </div>
    
  
    <div
      class="row clearfix"
    >
      
    
      <div
        class="col_half"
      >
        
      
        <div
          class="btn google"
          id="authGoogle"
        >
          <a
            href="#"
          >
            <span>
              <i
                aria-hidden="true"
                class="fab fa-google"
              />
            </span>
            Entrar com Google
          </a>
        </div>
        
      
        <div
          class="btn rg"
        >
          <a
            href="/register"
          >
            <span>
              <i
                aria-hidden="true"
                class="fas fa-user-circle"
              />
            </span>
            Criar nova conta
          </a>
        </div>
        
    
      </div>
      
    
      <div
        class="col_half last"
      >
        
      
        <form>
          
        
          <h3
            class="error"
            id="msgError"
          />
          
        
          <div
            class="input_field"
          >
            <span>
              <i
                aria-hidden="true"
                class="far fa-envelope"
              />
            </span>
            
          
            <input
              id="email-input"
              name="email"
              placeholder="Digite seu e-mail"
              required=""
              type="email"
            />
            
        
          </div>
          
        
          <div
            class="input_field"
          >
            <span>
              <i
                aria-hidden="true"
                class="fa fa-lock"
              />
            </span>
            
          
            <input
              id="password-home"
              placeholder="Digite sua senha"
              required=""
              type="password"
            />
            
        
          </div>
          
        
          <div>
            
          
            <input
              class="button"
              id="submit-home"
              type="submit"
              value="Entrar"
            />
             
        
          </div>
          
    
        </form>
      </div>
      
    
  
    </div>
    

  </div>
  


</div>
`;
